bigbluebutton-html5/imports/startup/client/auth.js

```diff
@@ -29,43 +33,43 @@ export function joinRouteHandler(nextState, replace, callback) {
         externUserID, fullname, confname, customdata,
       } = response;
 
+      console.log({ returncode });
       if (returncode === 'FAILED') {
-        replace({ pathname: '/error/404' });
-        callback();
+        Session.set('hasError', true);
+        Session.set('codeError', '404');
+        callback('failed unhappily');
+      } else {
+        setCustomLogoUrl(customLogoURL);
+
+        if (customdata.length) {
+          makeCall('addUserSettings', meetingID, internalUserID, customdata);
+        }
+
+        Auth.set(
+          meetingID, internalUserID, authToken, logoutUrl,
+          sessionToken, fullname, externUserID, confname,
+        );
+
+        Session.set('isUserListOpen', deviceInfo.type().isPhone);
+        const userInfo = window.navigator;
+
+        // Browser information is sent once on startup
+        // Sent here instead of Meteor.startup, as the
+        // user might not be validated by then, thus user's data
+        // would not be sent with this information
+        const clientInfo = {
+          language: userInfo.language,
+          userAgent: userInfo.userAgent,
+          screenSize: { width: window.screen.width, height: window.screen.height },
+          windowSize: { width: window.innerWidth, height: window.innerHeight },
+          bbbVersion: Meteor.settings.public.app.bbbServerVersion,
+          location: window.location.href,
+        };
+
+        logger.info(clientInfo);
+
+        callback('all is cool'); // TODO 4767
```

Comment:
the expected params are `(value, error)` you probably should change the callback call or the parameters to reflect the return.